.github/workflows/client-codegen.yaml

```diff
@@ -0,0 +1,33 @@
+name: GraphQL Client Codegen
+
+on:
+  push:
+    branches: ["**"]
+
+jobs:
+  codegen:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 10.10.0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+          cache: "pnpm"
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Run GraphQL codegen
```

Comment:
The workflow runs `pnpm client codegen` and `pnpm client build` but there's no validation that these scripts actually exist in package.json. If these scripts are missing, the workflow will fail. Consider adding error handling or validation that the required scripts exist.